### PR TITLE
Event Re-sender

### PR DESCRIPTION
**Pull request by RPeschke**
New functionality: Event Re-sender
This new re-sender can open old data files and re-sends them back to the EUDAQ system. It registers itself as Producer. The name will be the name from the sub-event-type. 

This pull request includes the old pull request.

### CODE REVIEW

File: debug/startrun.bat [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+taskkill /f /im TestDataCollector.exe.exe 
+taskkill /f /im ExampleProducer.exe.exe 
+
+start TestRunControl.exe.exe -a -c nim,ni_autotrig -d 100 -r 2
+Timeout /t 1 
+start TestLogCollector.exe.exe 
+Timeout /t 1 
+start TestDataCollector.exe.exe 
+Timeout /t 5 
+start ExampleProducer.exe.exe -r tcp://192.168.56.1:44000
```
Comment (simonspa) [inline debug/startrun.bat]: Hm, shall we - at some point - fix this double `.exe.exe` thing? It really looks ugly and unprofessional...

File: main/lib/src/TransportTCP.cc [hunk 2]
```diff
@@ -593,6 +628,7 @@ namespace eudaq {
             EUDAQ_THROW_NOLOG(LastSockErrorString("SocketClient Error (" + to_string(LastSockError()) + ")"));
           }
           else if (result > 0){
+			
             m_buf.append(result, buffer);
             while (m_buf.havepacket()) {
               m_events.push(TransportEvent(TransportEvent::RECEIVE, m_buf, m_buf.getpacket()));
```
Comment (simonspa) [inline main/lib/src/TransportTCP.cc]: Thanks, that doesn't belong into this PR, but I will pick it and push to v1.4-dev.

**Comment by simonspa** [comment]
One more general comment on this: currently the resender lives in the root directory which is not correct. It should go to `producers/resender`. Also I'm not a fan of cAmeL case - so better rename it to `resender` instead of `reSender`.

And finally: should this PG be filed against `v1.4-dev`? I completely lost track of `v2.0-dev`...

**Comment by simonspa** [comment]
Please also see comments by @hperrey in #59 - most of them still apply to this PR.

File: main/exe/src/TestRunControl.cxx [hunk 3]
```diff
@@ -24,18 +28,84 @@ class TestRunControl : public eudaq::RunControl {
...
 };
 
+struct rcHandler
+{
+  rcHandler(TestRunControl *rc, const std::vector<std::string>& conf, uint64_t duration, uint64_t runs) :m_rc(rc), m_conf(conf), m_duration(duration), m_runs(runs){}
+  TestRunControl *m_rc;
+  std::vector<std::string> m_conf;
+  uint64_t m_duration, m_runs;
+};
+
```
Comment (simonspa) [inline main/exe/src/TestRunControl.cxx]: does not belong to this PR (is for automated run control)

File: main/exe/src/TestRunControl.cxx [hunk 4]
```diff
@@ -24,18 +28,84 @@ class TestRunControl : public eudaq::RunControl {
...
 int main(int /*argc*/, const char ** argv) {
   eudaq::OptionParser op("EUDAQ Run Control", "1.0", "A command-line version of the Run Control");
   eudaq::Option<std::string> addr (op, "a", "listen-address", "tcp://44000", "address",
       "The address on which to listen for connections");
   eudaq::Option<std::string> level(op, "l", "log-level",      "NONE", "level",
       "The minimum level for displaying log messages locally");
+  eudaq::OptionFlag automatic(op, "a", "automatic", "runs the test run control in automatic mode");
+  eudaq::Option<std::string> configName(op, "c", "cnf", "", "configuration",
+    "determines which configuration gets loaded automatically");
+  eudaq::Option<uint64_t> duration(op, "d", "duration", 60, "uint64_t", "determines the length of a run in seconds");
```
Comment (simonspa) [inline main/exe/src/TestRunControl.cxx]: same as above: automated RC

File: main/exe/src/TestRunControl.cxx [hunk 5]
```diff
@@ -24,18 +28,84 @@ class TestRunControl : public eudaq::RunControl {
...
     bool done = false;
     bool help = true;
+
+    if (automatic.IsSet())
+    {
+      
+      eudaq::mSleep(10000); //30 seconds sleep to give the producer time to startup
+ 
+     
+
```
Comment (simonspa) [inline main/exe/src/TestRunControl.cxx]: same as above. automated RC

File: main/include/eudaq/FileWriter.hh [hunk 6]
```diff
@@ -19,6 +21,19 @@ namespace eudaq {
...
+	{
+		//std::cout<<"set output pattern to: "<<tagValue<<std::endl;
+		writer.SetFilePattern(tagValue);
+	}
+	  
+  }
+  inline void helper_ProcessEvent(FileWriter& writer,const DetectorEvent &ev){writer.WriteEvent(ev);}
+  inline void helper_StartRun(FileWriter& writer,unsigned runnumber){writer.StartRun(runnumber);}
+  inline void helper_EndRun(FileWriter& writer){};
+
```
Comment (simonspa) [inline main/include/eudaq/FileWriter.hh]: Multiple data collectors branch?

File: main/include/eudaq/readAndProcessDataTemplate.h [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      return true;
+    }
+    return false;
+  }
+  std::shared_ptr<processClass> m_processor;
+  eudaq::multiFileReader m_reader;
+  std::vector<unsigned> m_eventsOfInterest;
+};
+
+#endif // readAndProcessDataTemplate_h__
```
Comment (simonspa) [inline main/include/eudaq/readAndProcessDataTemplate.h]: MultiFileReader

**Comment by arummler** [comment]
Closing for rename of base branch as discussed per e-mail.
